src/main/java/de/tum/in/www1/artemis/service/metis/ConversationService.java

```diff
@@ -217,20 +218,44 @@ ZonedDateTime auditConversationReadTimeOfUser(Conversation conversation, User us
     }
 
     /**
-     * Update unreadMessageCount field of ConversationParticipant
+     * Increment unreadMessageCount field of ConversationParticipant
      *
      * @param userId            userId of the sender of the message(Post)
-     * @param courseId          courseId id of the course that the messaging is happening
      * @param conversationId    conversationId id of the conversation with participants
      */
-    public void unreadMessages(Long userId, Long courseId, Long conversationId) {
+    public void incrementUnreadMessagesCountOfParticipants(Long userId, Long conversationId) {
 
         List<ConversationParticipant> conversationParticipants = conversationParticipantRepository.findConversationParticipantByConversationId(conversationId);
-        ConversationParticipant receiverConversationParticipants = conversationParticipants.stream()
-                .filter(conversationParticipant -> conversationParticipant.getUser().getId() != userId).findAny().get();
-        long unreadMessagesCount = conversationRepository.findUnreadMessages(courseId, receiverConversationParticipants.getUser().getId(), conversationId);
+        conversationParticipants.forEach((conversationParticipant) -> {
```

Comment:
Thanks for the hint. I changed it to JPA query update. Maybe with 2 participants there is not much of a big difference in terms of performance but yes with 1000+ users it will make a difference.